src/App.tsx

```diff
@@ -1,61 +1,123 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './components/types/Todo';
+import React, { ChangeEvent, useState } from 'react';
+
+function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const getPreparedTodos = () => {
+  return todosFromServer.map(todo => ({
+    ...todo,
+    user: getUserById(todo.userId) || null,
+  }));
+};
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(getPreparedTodos());
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [titleInputError, setTitleInputError] = useState(false);
+  const [userSelectError, setUserSelectError] = useState(false);
+
+  const reset = () => {
+    setTitle('');
+    setSelectedUserId(0);
+    setTitleInputError(false);
+    setUserSelectError(false);
+  };
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    setTitleInputError(!title);
+    setUserSelectError(!selectedUserId);
+
+    if (!title && !selectedUserId) {
+      return;
+    }
+
+    setTodos(currentTodos => {
+      const newId = Math.max(0, ...currentTodos.map(todo => todo.id)) + 1;
+      const user = getUserById(selectedUserId);
```

Comment:
The `getUserById` function might return `null` if no user is found. Ensure that this case is handled properly when creating a new todo, as it might lead to unexpected behavior if `user` is `null`.